### PR TITLE
Create calendar and implement event functionality

### PR DESCRIPTION
**Pull request by Monica-Coira**
## Description

- Creates a calendar using React Big Calendar, which is a calendar component made for React, and Moment.js, which is a Javascript library that handles date and time operations.
- Implements Add Event and Edit Event functionality where the user can add or edit a class in their calendar.
- Implements four different views for the calendar: Month, Week, Day, and Agenda.
- Adds a "components" folder that houses all of my app's components to improve file organization.
- In the next PR, I will save the calendar events that the user creates in the database so that they can be used for the availability algorithm later on.

## Milestones
This works towards Milestone 5, which is that users can input class schedule/availability.

## Resources
I used React Big Calendar documentation (https://jquense.github.io/react-big-calendar/examples/?path=/docs/about-big-calendar--page) and a React Big Calendar Tutorial (https://www.youtube.com/watch?v=GrZHfeF8wEY) to help implement my calendar component.

## Test Plan

https://github.com/user-attachments/assets/3f65e019-374f-4b33-9527-a1044b743cf2

I tested adding events in different calendar views, on different days, at different times, and at overlapping times to ensure that the events were being properly created. I also tested editing events to ensure that they were being properly updated on the screen.
Besides testing basic functionality, these are the corner cases I’ve tested for the calendar:

- Edited an event’s title and time in the Month, Week, and Day view to ensure that the event is properly updated.
- Created an event that lasts all day to ensure that it appears in the All Day section at the top of the calendar.
- Clicked all blank cells and existing event cells to ensure that the correct modal pops up: either “Add Event” or “Edit Event”.
- Created 4+ events that all overlap to ensure that they are all visibly displayed and in the correct time slots.
- Left the Title, Start, and End input fields empty, alternating between which is empty, to ensure that the user will get an error message if they don’t fill out all three input fields.

### CODE REVIEW

**Comment by alisahq** [comment]
Great test plan! Would be helpful if you could list out all corner cases you've tested for the calendar views and times.

**Comment by alisahq** [comment]
qq - where do we store the actual calendar time? Are we planning to do so in the following commits?

File: client/src/components/EventPopup.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                        />
+                    </div>
+                    <div>
+                        <label>End:</label>
+                        <input
+                            className="event-popup-inputs"
+                            type="datetime-local"
+                            value={end}
+                            onChange={(event) => setEnd(event.target.value)}
+                            required
```
Comment (alisahq) [inline client/src/components/EventPopup.jsx:73]: nit - store these constant to a constant variable? and feel free to update all the magic numbers in the file.

File: client/src/components/PersonalityQuiz.jsx [hunk 2]
```diff
@@ -42,6 +44,9 @@ const PersonalityQuiz = () => {
             response: sliderValue
         }
         const newResponse = await fetchData("quiz/responses/", "POST", {"Content-Type": "application/json"}, "same-origin", JSON.stringify(newResponseData));
+        if (currentObject.id >= 10){
```
Comment (alisahq) [inline client/src/components/PersonalityQuiz.jsx:47]: nit - magic number?

File: client/src/components/EventPopup.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    const convertTimeZone = (date) => {
+        date.setHours(date.getHours() - 7);
+        return date;
+    }
+
+    useEffect(() => {
+        if (event) {
+            setTitle(event.title || "");
+            setStart(event.start.toISOString().slice(0, 16));
```
Comment (alisahq) [inline client/src/components/EventPopup.jsx:17]: qq - what does 16 mean?

**Review by alisahq — APPROVED** [review]
lgtm overall. Left a few questions.

File: client/src/components/PersonalityQuiz.jsx (hunk 2 above)
Comment (Monica-Coira) [reply client/src/components/PersonalityQuiz.jsx:47]: Got it, just changed it to a constant variable in my next PR.

File: client/src/components/EventPopup.jsx (hunk 1 above)
Comment (Monica-Coira) [reply client/src/components/EventPopup.jsx:73]: Got it, just updated these constants and all magic numbers to constant variables in the next PR.
